buildpack/layermetadata_test.go

```diff
@@ -33,12 +45,23 @@ func testLayerMetadata(t *testing.T, when spec.G, it spec.S) {
 			h.AssertNil(t, err)
 
 			var lmf buildpack.LayerMetadataFile
-			lmf, s, err := buildpack.DecodeLayerMetadataFile(metadataFile.Name(), "0.9")
+			lmf, err = buildpack.DecodeLayerMetadataFile(metadataFile.Name(), "0.9", logger)
 			h.AssertNil(t, err)
-			h.AssertEq(t, s, "")
 			h.AssertEq(t, lmf.Cache, true)
 			h.AssertEq(t, lmf.Build, false)
 			h.AssertEq(t, lmf.Launch, false)
 		})
+		it("metadata file in wrong format", func() {
```

Comment:
```suggestion
		it("logs a warning when the metadata file has wrong format (on older apis)", func() {
```

Reply:
We can make another `it` block for `returns an error when the metadata file has wrong format` that provides a newer buildpack API on the equivalent of line 59